doc/contributing.rst

```diff
@@ -0,0 +1,812 @@
+.. _contributing:
+
+**********************
+Contributing to xarray
+**********************
+
+.. contents:: Table of contents:
+   :local:
+
+.. note::
+
+  Large parts of this document came from the `Pandas Contributing
+  Guide <http://pandas.pydata.org/pandas-docs/stable/contributing.html>`_.
+
+Where to start?
+===============
+
+All contributions, bug reports, bug fixes, documentation improvements,
+enhancements, and ideas are welcome.
+
+If you are brand new to xarray or open-source development, we recommend going
+through the `GitHub "issues" tab <https://github.com/pydata/xarray/issues>`_
+to find issues that interest you. There are a number of issues listed under
+`Documentation <https://github.com/pydata/xarray/issues?q=is%3Aissue+is%3Aopen+label%3Adocumentation>`_
+and `good first issue
+<https://github.com/pydata/xarray/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22>`_
+where you could start out. Once you've found an interesting issue, you can
+return here to get your development environment setup.
+
+Feel free to ask questions on the `mailing list
+<https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!forum/xarray>`_ or on `Gitter`_.
```

Comment:
We previously dropped a gitter support in #1673 (or just removed a link from README), due to our limited resource.
I think we can remve this.